src/Object/AbstractObject.php

```diff
@@ -425,11 +448,11 @@ public function getFactor()
      *
      * @param string $value
      * @return self Provides a fluent interface
-     * @throws \Laminas\Barcode\Object\Exception\ExceptionInterface
+     * @throws ExceptionInterface
      */
     public function setForeColor($value)
     {
-        if (preg_match('`\#([0-9A-F]{6})`', $value, $reg)) {
+        if (preg_match('`\#([0-9A-F]{6})`', (string) $value, $reg)) {
```

Comment:
Ugly but understandable

Reply:
PHP 8.1: Deprecated function: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated
I can replace for Null coalescing operator `$value ?? ''`

Reply:
No, it's fine as-is: my comment was about the trade-off being taken